pkg/manager/member/tidb_init_manager.go

```diff
@@ -0,0 +1,381 @@
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: jobName,
+				},
+				Items: []corev1.KeyToPath{{Key: initStartKey, Path: initStartScriptPath}},
+			},
+		},
+	})
+	if ti.Spec.PasswordSecret != nil {
+		vms = append(vms, corev1.VolumeMount{
+			Name: passwdKey, ReadOnly: true, MountPath: passwdPath,
+		})
+		vs = append(vs, corev1.Volume{
+			Name: passwdKey,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: *ti.Spec.PasswordSecret,
+				},
+			},
+		})
+	}
+	if ti.Spec.InitSqlConfigMap != nil {
+		vms = append(vms, corev1.VolumeMount{
+			Name: sqlKey, ReadOnly: true, MountPath: sqlDir,
+		})
+		vs = append(vs, corev1.Volume{
+			Name: sqlKey,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: *ti.Spec.InitSqlConfigMap,
+					},
+					Items: []corev1.KeyToPath{{Key: sqlKey, Path: sqlPath}},
+				},
+			},
+		})
+	} else if ti.Spec.InitSql != nil {
+		vms = append(vms, corev1.VolumeMount{
+			Name: sqlKey, ReadOnly: true, MountPath: sqlDir,
+		})
+		vs = append(vs, corev1.Volume{
+			Name: sqlKey,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: jobName,
+					},
+					Items: []corev1.KeyToPath{{Key: sqlKey, Path: sqlPath}},
+				},
+			},
+		})
+	}
+
+	meta, initLabel := getInitMeta(ti)
+
+	podSpec := &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: initLabel.Labels(),
+		},
+		Spec: corev1.PodSpec{
+			InitContainers: []corev1.Container{
+				{
+					Name:    initContainerName,
+					Image:   v1alpha1.DefaultHelperImage,
+					Command: initcmds,
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      initStartKey,
+							ReadOnly:  true,
+							MountPath: path.Join(startScriptDir, initStartScriptPath),
+							SubPath:   initStartScriptPath,
+						},
+					},
+					Env: envs,
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:         containerName,
+					Image:        ti.Spec.Image,
+					Command:      cmds,
+					VolumeMounts: vms,
+					Env:          envs,
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+			Volumes:       vs,
+		},
+	}
+
+	if ti.Spec.ImagePullPolicy != nil {
+		podSpec.Spec.Containers[0].ImagePullPolicy = *ti.Spec.ImagePullPolicy
+	}
+	if ti.Spec.Resources != nil {
+		podSpec.Spec.Containers[0].Resources = *ti.Spec.Resources
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: meta,
+		Spec: batchv1.JobSpec{
+			BackoffLimit: controller.Int32Ptr(0),
+			Template:     *podSpec,
+		},
+	}
+
+	return job, nil
+}
+
+func getTiDBInitConfigMap(ti *v1alpha1.TidbInitializer) (*corev1.ConfigMap, error) {
+	var permitHost string
+	var initSQL, passwdSet bool
+
+	if ti.Spec.PermitHost == nil || *ti.Spec.PermitHost == "" {
+		permitHost = "%"
+	} else {
+		permitHost = *ti.Spec.PermitHost
+	}
```

Comment:
How about adding a `GetPermitHost()` method to `TidbInitializer` struct so that the default value could be hidden

Reply:
updated